packit/upstream.py

```diff
@@ -921,3 +920,103 @@ def get_version_from_tag(self, tag: str) -> str:
             )
             logger.error(msg)
             raise PackitException(msg)
+
+    def get_archive_root_dir(self, archive: str) -> Union[str, None]:
+        """
+        Returns archive root dir.
+        It uses 2 techniques:
+        1. tries to extract it directly from archive.
+        2. will generate name based on archive_root_dir_template
+
+        Currently supported archives:
+        * tar including compression (for details check python tarfile module doc)
+
+        :param archive: name of archive
+        :return: archive top level directory or None
+
+        .. raises:: PackitException if failed - all used methods returned None
+        """
+
+        archive_root_dir = None
+
+        # 1. try extract from archive
+        if tarfile.is_tarfile(f"{self.absolute_specfile_dir}/{archive}"):
+            logger.debug(f"Archive {archive} is tar.")
+            archive_root_dir = self.get_archive_root_dir_from_tar(archive)
+
+        # 2. try to use archive_root_dir_template to generate correct dir
+        if archive_root_dir is None:
+            archive_root_dir = self.get_archive_root_dir_from_template()
+
+        if archive_root_dir is None:
+            raise PackitException("Unable to get archive root dir.")
```

Comment:
I would like to include a few points here since it will be probably shown to the user:
- The dir is searched for tar files.
- The default scheme is used otherwise.
- Users can redefine that by setting `archive_root_dir_template` config value.

Reply:
Just realize that also archive_root_dir will be never None because archive_root_dir_template will be always a valid string as you pointed in another part. So I removed this section and improved logging so a user is informed whats happening. 

WDYT?